### ISSUE TITLE
Missing HTML widget attributes in StreamField widget rendering

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#10300]
### Issue Summary

Comparing the rendering of widgets in StreamField and outside, there are a number of attributes missing in the StreamField (Telepath?) variant. This is likely because we added those attributes to the FieldPanel’s [custom rendering](https://github.com/wagtail/wagtail/blob/main/wagtail/admin/panels/field_panel.py#L175) only, and didn’t replicate it for fields outside FieldPanel.

Specific attributes that are missing:

- [x] `required` for required fields. This likely predates any forms changes in v4.0
- [x] `aria-invalid="true"` for fields with errors
- [x] `aria-describedby` with an ID list pointing at any help text
- [ ] `aria-describedby` with an ID list pointing at any error messages

All `input type="text"` in StreamField seem to only ever render as:

```html
<input type="text" name="backstory-0-value-heading_text" id="backstory-0-value-heading_text">
```

Similarly a required `textarea` with an error message is:

```html
<textarea name="backstory-2-value-text" cols="40" rows="1" data-controller="w-autosize" class="w-field__autosize" id="backstory-2-value-text" style="overflow: hidden; overflow-wrap: break-word; text-align: start; height: 71px;"></textarea>
```

For comparison, here’s the markup for our page `title` field (excluding Stimulus attributes):

```html
<input type="text" name="title" placeholder="Page title*" maxlength="255" aria-invalid="true" aria-describedby="panel-child-content-child-title-helptext panel-child-content-child-title-errors" required="" id="id_title">
```

### Steps to Reproduce

1. Go to the homepage edit UI in the bakerydemo
2. In the Body field, add a new Embed block without setting a URL
3. Save draft
4. Inspect the URL input field for the block. It will be: `<input type="url" name="body-0-value" id="body-0-value">`

Instead, it should have the attributes mentioned above.

**Comment by Pratik-A-N** [comment]
Hi, can you assign this issue to me ??

**Comment by lb-** [comment]
I wonder if this has been fixed via https://github.com/wagtail/wagtail/pull/10195 & https://github.com/wagtail/wagtail/issues/10170

**Comment by thibaudcolas** [comment]
Assigning to @Stormheg. I suspect adding those attributes would need to be done either in the FieldBlock rendering of the widget, here: https://github.com/wagtail/wagtail/blob/main/client/src/components/StreamField/blocks/FieldBlock.js#L43-L49

Or (more likely I think) where we render the actual widget: https://github.com/wagtail/wagtail/blob/69724e4e3a3d372fa3cd47c66c11df74daa203bb/client/src/entrypoints/admin/telepath/widgets.js#L52

**Comment by gasman** [comment]
Sounds like this might be the cause of #10640 too...?

**Comment by thibaudcolas** [comment]
Yes! Will add details accordingly, good spotting.

**Comment by Stormheg** [comment]
PR up for adding `required` and `aria-describedby` for help text: https://github.com/wagtail/wagtail/pull/10705

**Comment by lb-** [comment]
https://github.com/wagtail/wagtail/pull/10705 fixed the `required` and `aria-describedby` but the aria-invalid is still to be done, hence re-opening issue.

**Comment by thibaudcolas** [comment]
Almost there with #10705 and now #13204. The only change that is still needed is:

> `aria-describedby` with an ID list pointing at any error messages

This is simplest to test with Embed blocks. Here is how the embed URL input renders when there is both help text and an error:

```html
<input type="url" name="body-2-value" id="body-2-value" aria-describedby="body-2-value-helptext" required="" aria-invalid="true">
```

The `aria-describedby` should instead contain both, ordered matching the DOM: `body-2-value-errors body-2-value-helptext`. And if there is an error and no help text, `aria-describedby` should be the ID of the error container element. This can be tested with Heading blocks’ heading text in the bakery demo.

---

To implement this, I suspect the simplest will be to update the widget `setInvalid` method implemented in #13204 to the error container’s ID as a parameter. Then widgets can set their `aria-describedby` attribute according to the presence of that ID, alongside `aria-invalid`.
